### PR TITLE
[Notifi] notification popover integration v1.2

### PR DESCRIPTION
**Pull request by eric-notifi**
## What is the purpose of the change

Integrate notification popover modal powered by notifi.network

### 8/3

- Rebase and Fix conflicts

### 8/1

- Notifi popover button state optimization

### 7/31

- Fix the conflicts
- Re-enable "Positions out of range" alert

### 7/27
Fix merge stage confliects

### 7/25
- Change Positions out of range type defs
- Adopt tailwind style to dummy notification history

### 7/24
Fix the notificationHistory fetch fail err message

### 7/22
Resolve stage conflict

### 7/21
Fix merge stage conflicts

### 7/20 update 
- Squash commits
- Rebase master (Fix conflict)
- Notifi card localization
- New Notifi alert icons
- Include prod config
- import statement
- Disable and remove “Transaction status” from `alert-list`

### 7/19 PR update

- style (sizing with rem; text/font category/ Align dropdown Panel and icon button/ History row timestamp format)
- Create `useNotifiBreadcrumb` hook
- Add env variable to distinguish dev and prod
- Rebase master (No conflict)
- Fix deposit and trading urls

### 7/18 update (New branch and PR for rebasing master and avoiding conflict)
- Squash commits
- Rebase master
- Add breadcrumb
- Optimize style (pixel precise)
- Add hover states

### CODE REVIEW

File: packages/web/integrations/notifi/index.tsx
Comment (jonator) [inline packages/web/integrations/notifi/index.tsx:1]: Let's convert this to a .ts file

File: packages/web/integrations/notifi/notifi-context.tsx [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+            info.chainId,
+            info.account.address,
+            message
+          );
+          return Buffer.from(result.signature, "base64");
+        }}
+      >
+        <NotifiConfigContext
+          type="SUBSCRIPTION_CARD"
+          id="c5fb30811f5b47b79dd9a400480c2670"
```
Comment (jonator) [inline packages/web/integrations/notifi/notifi-context.tsx:79]: Do you guys have a testnet environment? If so, let's use IS_TESTNET to use that. If you don't have an env, I think you should for testing.

File: packages/web/integrations/notifi/notifi-popover.tsx [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+        <Popover.Button as={Fragment}>
+          <NotifiIconButton
+            className={className}
+            hasUnreadNotification={hasUnreadNotification}
+          />
+        </Popover.Button>
+        <Popover.Panel
+          className={classNames(
+            "absolute bottom-[-0.5rem] right-[-10rem] z-40",
+            `h-[640px] w-[420px]`,
```
Comment (jonator) [inline packages/web/integrations/notifi/notifi-popover.tsx:90]: How does this work on mobile? Has mobile been considered?

File: packages/web/integrations/notifi/notifi-subscription-card/fetched-card/alert-list.tsx [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+  }
+
+  return (
+    <ul className="mt-[-18px] block px-[40px] pb-[18px] text-[12px] font-[500]">
+      {config.data.eventTypes.map((row) => {
+        if (row.type === "label") {
+          return (
+            <li
+              key={row.name}
+              className="mb-[24px] mt-[36px] flex flex-col gap-[8px]"
```
Comment (jonator) [inline packages/web/integrations/notifi/notifi-subscription-card/fetched-card/alert-list.tsx:28]: I see absolute pixels used everywhere, is it possible to use REM anywhere? That's our convention in our app.

File: packages/web/integrations/notifi/notifi-subscription-card/fetched-card/history-rows.tsx [hunk 4]
```diff
@@ -0,0 +1,432 @@
...
+        day.isAfter(dayjs(Date.now()).subtract(2, "day")) && !isToday;
+      const timestamp = isToday
+        ? day.format("h:mm A")
+        : isYesterday
+        ? day.format("h:mm A") + " Yesterday"
+        : day.format("MMMM D, YYYY h:mm A");
+
+      const rowProps = {
+        emoji: "😵",
+        title: "Unsupported Notification",
```
Comment (jonator) [inline packages/web/integrations/notifi/notifi-subscription-card/fetched-card/history-rows.tsx:60]: Will you guys have localizations? I'd add them to a "notifi" member of our translations file en.json. Then, you should be able to run `yarn machine-translate` to generate initial translations automatically.

File: packages/web/localizations/en.json [hunk 5]
```diff
@@ -340,6 +340,9 @@
     "vote": "Vote",
     "new": "NEW"
   },
+  "notifi": {
+    "title": "Notifications"
```
Comment (jonator) [inline packages/web/localizations/en.json:344]: Let's add the remaining strings here, to be translated.

File: packages/web/integrations/notifi/notifi-popover.tsx (hunk 2 above)
Comment (eric-notifi) [reply packages/web/integrations/notifi/notifi-popover.tsx:90]: Yeah, Mobile is also considered. 
The `NotifiPopover`  container is used for desktop view. 
The mobile view adopts `NotifiModal` is used for for small view

File: packages/web/integrations/notifi/notifi-subscription-card/fetched-card/history-rows.tsx (hunk 4 above)
Comment (eric-notifi) [reply packages/web/integrations/notifi/notifi-subscription-card/fetched-card/history-rows.tsx:60]: Thanks! Will use it with some constant values.
In this case of `rowProps`, It is dynamic value though. we might not be able to adopt in this block.

File: packages/web/integrations/notifi/notifi-context.tsx (hunk 1 above)
Comment (eric-notifi) [reply packages/web/integrations/notifi/notifi-context.tsx:79]: Is it ok for me to add new variables in .env . Like below
```
NEXT_PUBLIC_NOTIFI_CARD_ID_TESTNET=xxx
NEXT_PUBLIC_NOTIFI_CARD_ID_PRODUCTION=xxx
```
or you will prefer put the id in-line.

File: packages/web/components/navbar/index.tsx [hunk 6]
```diff
@@ -62,6 +77,41 @@ export const NavBar: FunctionComponent<
...
+          const newestNotificationCreatedDate = newestHistoryItem?.createdDate
+            ? dayjs(newestHistoryItem?.createdDate)
+            : dayjs("2022-01-05T12:30:00.792Z");
+
+          const lastStoredTimestamp = dayjs(
+            window.localStorage.getItem(localStorageKey)
+          ).isValid()
+            ? dayjs(window.localStorage.getItem(localStorageKey))
+            : dayjs("2022-01-05T10:30:00.792Z");
+
```
[the remark this answers is not in the document]
Comment (eric-notifi) [reply packages/web/components/navbar/index.tsx]: Created `useNotifiBreadcrumb` hook in `integrations/notifi/hook`

File: packages/web/integrations/notifi/notifi-context.tsx (hunk 1 above)
Comment (eric-notifi) [reply packages/web/integrations/notifi/notifi-context.tsx]: For now the updated PR is using in-line. can change to env anytime.

File: packages/web/integrations/notifi/notifi-subscription-card/fetched-card/alert-list.tsx (hunk 3 above)
Comment (eric-notifi) [reply packages/web/integrations/notifi/notifi-subscription-card/fetched-card/alert-list.tsx]: Updated with fix.

File: packages/web/integrations/notifi/notifi-subscription-card/fetched-card/history-detail-view.tsx [hunk 7]
```diff
@@ -0,0 +1,60 @@
...
+        message = "Oops, something went wrong. Please let us know";
+        break;
+    }
+    return { title, timestamp, message };
+  }, [historyRowData]);
+  return (
+    <>
+      <div className=" flex flex-col px-[32px]">
+        <div className="mb-5 flex flex-row items-center justify-between">
+          <div className="font-600 col-span-2 text-[16px]">{title}</div>
```
[the remark this answers is not in the document]
Comment (eric-notifi) [reply packages/web/integrations/notifi/notifi-subscription-card/fetched-card/history-detail-view.tsx]: Updated with fix.

File: packages/web/integrations/notifi/notifi-subscription-card/index.tsx
[the remark this answers is not in the document]
Comment (eric-notifi) [reply packages/web/integrations/notifi/notifi-subscription-card/index.tsx:1]: Updated with fix.

File: packages/web/integrations/notifi/notifi-context.tsx (hunk 1 above)
Comment (jonator) [reply packages/web/integrations/notifi/notifi-context.tsx]: I think inline is fine for now

File: packages/web/integrations/notifi/notifi-subscription-card/fetched-card/history-rows.tsx (hunk 4 above)
Comment (eric-notifi) [reply packages/web/integrations/notifi/notifi-subscription-card/fetched-card/history-rows.tsx:60]: Oh ... It means that in this case the `rowProps`  content changes over time. 
each rendered component has its own corresponding content (`rowProps`).

Comment (jonator) [reply packages/web/integrations/notifi/notifi-subscription-card/fetched-card/history-rows.tsx:60]: And how does that prevent localization? I'd like to, if possible, avoid having unlocalized content along with localized content

**Review by jonator — COMMENTED** [review]
The most recommended way to work with SVGs in React is with sprite components. It has the best balance of performance and ability to modify SVG attributes. I'd recommend look at our Icon.tsx and consider creating your own sprite.svg and NotifiIcon.tsx to host these notifi icons.

Also heads up I changed the base branch to stage.

**Comment by eric-notifi** [comment]
After trying, the sprite components does not work with `<linearGradient>` svg element. 
But now the new icons are full-colored using `<linearGradient>` element. it will break the svg (render nothing) by adopting sprite.

**Review by jonator — APPROVED** [review]

**Comment by jonator** [comment]
@eric-notifi it still says there's conflicts with base branch. Though I can't see them. Can you please resolve them?

**Comment by eric-notifi** [comment]
@jonator  Conflict resolved

**Review by jonator — CHANGES_REQUESTED** [review]
Is this notification button disabled since my wallet is not connected?

Screenshot 2023-07-31 at 3 33 55 PM

If so, I think the hover style of the icon should also be disabled:
Screenshot 2023-07-31 at 3 34 24 PM

**Comment by eric-notifi** [comment]
PR updated (Button state optimization)

**Review by jonator — APPROVED** [review]

**Pull request merged**